src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import java.lang.Math;
+import com.revrobotics.REVLibError;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.SparkMaxRelativeEncoder;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+
+import org.opencv.core.Mat;
+import com.revrobotics.RelativeEncoder;
+import frc.robot.common.hardware.MotorController;
+
+public class ShooterSubsystem extends SubsystemBase {
+  private int aimMode; // 0 is LOW, 1 is AUTO, 2 is LAUNCH, 3 is TARMAC
+  private MotorController shooter_motorController;
+  private MotorController hood_motorController;
+  private SparkMaxPIDController KShooterController;
+  private SparkMaxPIDController KHoodController;
+  private RelativeEncoder KShooterEncoder;
+  private RelativeEncoder KHoodEncoder;
+  private MotorController cargo_motorController;
+  private SparkMaxPIDController kCargoController;
+  private RelativeEncoder kCargoEncoder;
+
+  public ShooterSubsystem() {
+
+    aimMode = 1;
+    cargo_motorController = new MotorController("Shooter Cargo", Constants.KShooterCargoID);
+    kCargoController = cargo_motorController.getPID();
+    kCargoEncoder = cargo_motorController.getEncoder();
+
+    shooter_motorController = new MotorController("Shooter", Constants.KShooterID);
+    KShooterController = shooter_motorController.getPID();
+    KShooterEncoder = shooter_motorController.getEncoder();
+    hood_motorController = new MotorController("Hood", Constants.KHoodID);
+    KHoodController = hood_motorController.getPID();
+    KHoodEncoder = shooter_motorController.getEncoder();
+
+  }
+
+  public void adjustHood(double a) {
+    KHoodController.setReference(a, CANSparkMax.ControlType.kPosition);
+
+    // Adjusts Hood using PID control to passed angle a
+  }
+
+  public void windFlywheel(int rpm) {
+
+    double kGearRationRPM = rpm * (Constants.KGearRatioIn / Constants.KGearRatioOut);
+    KShooterController.setReference(kGearRationRPM, CANSparkMax.ControlType.kVelocity);
+    while (true) {
```

Comment:
Instead of doing this 'blocking' check, create a separate method to check for ready once each 'execute' phase of your ShooterPrime command (or in 'isFinished'). Then have another method to run the CargoController motor in the 'end' phase. This prevents your command from blocking other scheduled commands while you wait for the shooter to be ready.